corebehrt/functional/cohort_handling/advanced/extract.py

```
@@ -70,8 +71,19 @@ def get_birth_date_for_each_patient(events: pd.DataFrame) -> pd.Series:
 @lru_cache(maxsize=128)
 def _compile_regex(patterns: tuple) -> re.Pattern:
-    """Cache compiled regex patterns."""
-    return re.compile("|".join(patterns))
+    """
+    Cache compiled regex patterns.
+    Wrap each pattern in a non-capturing group.
+    """
+    if len(patterns) == 1:
+        return re.compile(patterns[0])
+    if "?i" in patterns:
+        warnings.warn(
```

Ruff (0.11.9)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)